test_runner/regress/test_metric_collection.py

```diff
@@ -0,0 +1,73 @@
+import pytest
+from fixtures.log_helper import log
+from fixtures.neon_fixtures import NeonEnvBuilder, PortDistributor
+from pytest_httpserver import HTTPServer
+from werkzeug.wrappers.request import Request
+from werkzeug.wrappers.response import Response
+
+
+@pytest.fixture(scope="session")
+def httpserver_listen_address(port_distributor: PortDistributor):
+    port = port_distributor.get_port()
+    return ("localhost", port)
+
+
+num_metrics_received = 0
+
+
+#
+# verify that metrics look minilally sane
+#
+def metrics_handler(request: Request) -> Response:
+    if request.json is None:
+        return Response(status=400)
+
+    events = request.json["events"]
+    log.info("received events:")
+    log.info(events)
+
+    checks = {
+        "written_size": lambda value: value > 0,
+        "physical_size": lambda value: value >= 0,
+        "s3_storage_size": lambda value: value == 0,
+    }
+
+    for event in events:
+        assert checks.pop(event["metric"])(event["value"]), f"{event['metric']} isn't valid"
```

Comment:
This was some tricky python for me on the first view.

I'd be more happy if the checks were run against all values, not just the first ones. I understand the first found is popped off, ... but that can't be right, then the second "written_size" would fail the `checks.pop` with some `ValueError`? I don't think I understand this test case after all. There should be two tenants on the pageserver in this test, right? I don't understand how could only 3 metrics come out of 2 tenants with 1 timeline each.

Reply:
there's just one tenant in this test, so this is not too complicated.
also mock server only serves the first shot, so all metrics should be present here

Reply:
Ah oki, I misread that a new tenant would be created so there would be two (default and created) but that was only two timelines on 1 tenant. Got it, resolving.